libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -413,6 +417,19 @@ bool AP_BattMonitor::get_temperature(float &temperature, const uint8_t instance)
     }
 }
 
+bool AP_BattMonitor::arming_checks(size_t buflen, char *buffer) const
+{
+    char temp_buffer[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1] {};
```

Comment:
I considered it, but since it makes us really dependent upon how the caller prefixes the message it could break over time. The advantage of this is this is the maximum possible length we could emit, and since we never fill it anyways we only pay a small price to pay for pushing it on the stack. I can change it if it's a real issue though. I've just been bitten by hard coding assumption about prefix length and then forgetting to update it later.

Reply:
OK, I'll allow it 😛